app/code/Magento/InventoryCatalog/etc/di.xml

```diff
@@ -8,4 +8,7 @@
 <config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
     <preference for="Magento\InventoryCatalog\Api\DefaultStockProviderInterface" type="Magento\InventoryCatalog\Model\DefaultStockProvider"/>
     <preference for="Magento\InventoryCatalog\Api\DefaultSourceProviderInterface" type="Magento\InventoryCatalog\Model\DefaultSourceProvider"/>
+    <type name="Magento\Inventory\Model\StockRepository">
+        <plugin name="prevent-default-stock-deletion" type="Magento\InventoryCatalog\Plugin\Model\StockRepositoryPlugin" sortOrder="1"/>
```

Comment:
As I see in Magento generally used underscore for keys (but not dashe)

Reply:
got it, again like in magento2 original code but of course I will fix that